opencti-platform/opencti-front/src/private/components/common/files/workbench/WorkbenchFileCreator.tsx

```diff
@@ -0,0 +1,177 @@
+import TextField from '../../../../../components/TextField';
+import AutocompleteFreeSoloField from '../../../../../components/AutocompleteFreeSoloField';
+import ItemIcon from '../../../../../components/ItemIcon';
+import { useFormatter } from '../../../../../components/i18n';
+import { Theme } from '../../../../../components/Theme';
+import { Option } from '../../form/ReferenceField';
+import { WorkbenchFileCreatorMutation } from './__generated__/WorkbenchFileCreatorMutation.graphql';
+import { WorkbenchFileViewer_entity$data } from './__generated__/WorkbenchFileViewer_entity.graphql';
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  icon: {
+    paddingTop: 4,
+    display: 'inline-block',
+    color: theme.palette.primary.main,
+  },
+  text: {
+    display: 'inline-block',
+    flexGrow: 1,
+    marginLeft: 10,
+  },
+  autoCompleteIndicator: {
+    display: 'none',
+  },
+}));
+
+const workbenchFileCreatorMutation = graphql`
+  mutation WorkbenchFileCreatorMutation($file: Upload!, $labels: [String], $entityId: String) {
+    uploadPending(file: $file, labels: $labels, errorOnExisting: true, entityId: $entityId) {
+      id
+      ...FileLine_file
+    }
+  }
+`;
+
+const fileValidation = (t: (value: string) => string) => Yup.object().shape({
+  name: Yup.string().required(t('This field is required')),
+});
+
+interface WorkbenchFileCreatorFormValues {
+  name: string;
+  labels: Option[];
+}
+
+interface WorkbenchFileCreatorProps {
+  openCreate: boolean;
+  handleCloseCreate: () => void;
+  onCompleted?: () => void;
+  entity?: WorkbenchFileViewer_entity$data;
+}
+
+const WorkbenchFileCreator: FunctionComponent<WorkbenchFileCreatorProps> = ({ openCreate, handleCloseCreate, onCompleted, entity }) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+  const [commitWorkbench] = useMutation<WorkbenchFileCreatorMutation>(workbenchFileCreatorMutation);
+  const entityId = entity?.id;
+  const onSubmitCreate: FormikConfig<WorkbenchFileCreatorFormValues>['onSubmit'] = (
+    values,
+    { setSubmitting, resetForm },
+  ) => {
+    let { name } = values;
+    const finalLabels = values.labels.map((label) => label.value);
+    if (!name.endsWith('.json')) {
+      name += '.json';
+    }
+    const objects = [];
+    if (entity?.toStix) {
+      const stixEntity = JSON.parse(entity.toStix);
+      delete stixEntity.extensions;
+      stixEntity.x_opencti_id = entity.id;
+      objects.push(stixEntity);
+    }
+    const data = { id: `bundle--${uuid()}`, type: 'bundle', objects };
+    const json = JSON.stringify(data);
+    const blob = new Blob([json], { type: 'text/json' });
+    const file = new File([blob], name, {
+      type: 'application/json',
+    });
+
+    commitWorkbench({
+      variables: { file, labels: finalLabels, entityId },
+      onCompleted: () => {
+        setSubmitting(false);
+        resetForm();
+        handleCloseCreate();
+        if (onCompleted) {
```

Comment:
onCompleted?.();

Reply:
hum, I don't see this very often in the code, I'm not sure everyone will understand this syntax, the if is more understandable

Reply:
We don't see it often because it is rather new, but we want to use it, the same way as we use ?? or object?.property

Reply:
Done